pandera/backends/polars/container.py

```diff
@@ -25,11 +25,31 @@
 from pandera.validation_depth import validate_scope, validation_type
 
 
+def _to_lazy(df: PolarsFrame) -> pl.LazyFrame:
+    if isinstance(df, pl.DataFrame):
+        return df.lazy()
+    else:
+        return df
+
+
+def _to_frame_kind(obj: PolarsFrame, kind: type[PolarsFrame2]) -> PolarsFrame2:
+    if isinstance(obj, pl.DataFrame):
+        if issubclass(kind, pl.DataFrame):
```

Comment:
this could be if kind ==pl.DataFrame but mypy doesn't seem to narrow on that properly